lib/ansible/modules/network/checkpoint/cp_mgmt_threat_rule.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Ansible module to manage CheckPoint Firewall (c) 2019
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: cp_mgmt_threat_rule
+short_description: Manages threat-rule objects on Checkpoint over Web Services API
+description:
+  - Manages threat-rule objects on Checkpoint devices including creating, updating and removing objects.
+  - All operations are performed over Web Services API.
+version_added: "2.9"
+author: "Or Soffer (@chkp-orso)"
+options:
+  position:
+    description:
+      - Position in the rulebase.
+    type: str
+  layer:
+    description:
+      - Layer that the rule belongs to identified by the name or UID.
+    type: str
+  name:
+    description:
+      - Object name.
+    type: str
+    required: True
+  action:
+    description:
+      - Action-the enforced profile.
+    type: str
+  destination:
+    description:
+      - Collection of Network objects identified by the name or UID.
+    type: list
+  destination_negate:
+    description:
+      - True if negate is set for destination.
+    type: bool
+  enabled:
+    description:
+      - Enable/Disable the rule.
+    type: bool
+  install_on:
+    description:
+      - Which Gateways identified by the name or UID to install the policy on.
+    type: list
+  protected_scope:
+    description:
+      - Collection of objects defining Protected Scope identified by the name or UID.
+    type: list
+  protected_scope_negate:
+    description:
+      - True if negate is set for Protected Scope.
+    type: bool
+  service:
+    description:
+      - Collection of Network objects identified by the name or UID.
+    type: list
+  service_negate:
+    description:
+      - True if negate is set for Service.
+    type: bool
+  source:
+    description:
+      - Collection of Network objects identified by the name or UID.
+    type: list
+  source_negate:
+    description:
+      - True if negate is set for source.
+    type: bool
+  track:
+    description:
+      - Packet tracking.
+    type: str
+  track_settings:
+    description:
+      - Threat rule track settings.
+    type: list
```

Comment:
@chkp-orso This can be converted to `dict` instead of `list`
```suggestion
    type: dict
```